### ISSUE TITLE
Hamiltonian Neural Networks don't work on GPU

### ISSUE THREAD

**Issue by avik-pal** [SciML/DiffEqFlux.jl#378]
For HNN we need to supervise the network on its gradients which are computed by Zygote. Since we are currently unable to nest Zygote, we use ReverseDiff for this. This prevents training on GPUs. As pointed out in [here](https://github.com/SciML/DiffEqFlux.jl/pull/370#issuecomment-663846037) this is supposed to be fixed by Keno's AD.

Opening this issue to keep track of the progress on GPU support.
